src/adapter.js

```diff
@@ -331,12 +389,15 @@ class MongooseAdapter {
     if (fieldIndex <= 5 && fieldIndex + fieldValues.length > 5 && fieldValues[5 - fieldIndex]) {
       where.v5 = fieldValues[5 - fieldIndex];
     }
-
-    await CasbinRule.deleteMany(where);
+    const options = {};
+    if (this.useTransaction) options.session = await this.getTransaction();
+    await CasbinRule.deleteMany(where, options);
+    if (this.useTransaction && this.debug) console.log('Lines deleted in transaction, but it\'s not commited yet.');
```

Comment:
If you want to have debug logs, better to use `debug` package. You can find it on npm.

Reply:
Yeah, I was wondering if we should add some notification for new users that they actually need to commit their changes before they appear in database or whether code examples in readme are enough.

Thoughts?

Reply:
Personally, I would rather use just examples in README than including another dependancy. Some `node-casbin` packages use their own logger.

Reply:
`debug` is really small in size and plus, you give consumers of your library a possibility to debug adapter on their own side.

Reply:
> Yeah, I was wondering if we should add some notification for new users that they actually need to commit their changes before they appear in database or whether code examples in readme are enough.

in case you want to do that, you definitely need to write this warning without debug flag 😄 
most of the consumers will be working with it without debug flag and if you want users to see the message, you shouldn't check for it.

Reply:
`debug` is probably fine. I was just wondering if there are already some packages / implementations that are in use in other `node-casbin` packages. So we keep the implementations closely similar, so that others can work on them with ease. (ofc, debug package is a small addition, but just a thought.) 

In general though, should we put out some kind of warning? I just added the debug tag just so people can silence the "warnings" if they want. Shall we do this via `debug` or something else? I just did not add `debug` at the start since the use of it was quite limited or minimal and none of the other packages were not using it (or logging events to that matter).

I'm fine with everything 😁

Reply:
> debug is probably fine. I was just wondering if there are already some packages / implementations that are in use in other node-casbin packages. So we keep the implementations closely similar, so that others can work on them with ease. (ofc, debug package is a small addition, but just a thought.)

At least I don't heard about it, so I think you can freely use `debug` package here.

> In general though, should we put out some kind of warning? I just added the debug tag just so people can silence the "warnings" if they want. Shall we do this via debug or something else? I just did not add debug at the start since the use of it was quite limited or minimal and none of the other packages were not using it (or logging events to that matter).

Warnings if something is going wrong is always good, so if you think it makes sense, of course you can do it. Though, it must not go through `debug` module, maybe `console.warn()`, so any user with any configuration could see it.

Reply:
Added `console.warn`to transaction start and abort (since both can triggered without user interaction) and `console.error`to savePolicy's throw error. I do not think it should throw an error, just false/true wether action succeeded. IMO this is more in line with other Casbin implementations.